File: pocs/metrics-report-skill/sample-app/java-react/backend/src/main/java/com/retirement/model/YearlyProjection.java
package com.retirement.model;

public class YearlyProjection {

    private int year;
    private int age;
    private double startBalance;
    private double contributions;
    private double interestEarned;
    private double endBalance;

    public YearlyProjection(int year, int age, double startBalance, double contributions, double interestEarned, double endBalance) {
        this.year = year;
        this.age = age;
        this.startBalance = startBalance;
        this.contributions = contributions;
        this.interestEarned = interestEarned;
        this.endBalance = endBalance;
    }

    public int getYear() { return year; }
    public int getAge() { return age; }
    public double getStartBalance() { return startBalance; }
    public double getContributions() { return contributions; }
    public double getInterestEarned() { return interestEarned; }
    public double getEndBalance() { return endBalance; }
}

File: pocs/metrics-report-skill/sample-app/scala3-react/backend/src/main/scala/com/retirement/model/YearlyProjection.scala
package com.retirement.model

import scala.beans.BeanProperty

class YearlyProjection(
  @BeanProperty var year: Int,
  @BeanProperty var age: Int,
  @BeanProperty var startBalance: Double,
  @BeanProperty var contributions: Double,
  @BeanProperty var interestEarned: Double,
  @BeanProperty var endBalance: Double
)
